### ISSUE TITLE
Downloading omnibus-toolchain gives 'Unknown filetype'

### ISSUE THREAD

**Issue by jonathanmorley** [chef/mixlib-install#148]
https://github.com/chef/mixlib-install/compare/v2.0.0...v2.0.1#diff-45f7ab6089f4d31bac37bd31dd8b48caL49

The removal of the line above seems to have caused a regression where the omnibus-toolchain cookbook cannot verify the filetype of the downloaded package

```
default-jhcook-macos-sierra:~ vagrant$ /bin/sh  /tmp/kitchen/cache/installer.sh
mac_os_x 10.12 x86_64
Getting information for omnibus-toolchain stable 1.1.29 for mac_os_x...
downloading https://omnitruck.chef.io/stable/omnibus-toolchain/metadata?v=1.1.29&p=mac_os_x&pv=10.12&m=x86_64
  to file /var/folders/rm/ywx69srx1kq_dtjz2ynznn000000gp/T//install.sh.938/metadata.txt
trying curl...
sha1	56e7a3dc124d43ff23c90107984f16886a52ecf8
sha256	78bb008e18fec019fcf043b095c9a53acb97cd787adb15f34ddaebeaeea90089
url	https://packages.chef.io/stable/mac_os_x/10.11/omnibus-toolchain-1.1.29-1.dmg
version	1.1.29
downloaded metadata file looks valid...
Installing omnibus-toolchain 1.1.29
Unknown filetype:
Version: 1.1.29

Please file a Bug Report at https://github.com/chef/omnitruck/issues/new
Alternatively, feel free to open a Support Ticket at https://www.chef.io/support/tickets
More Chef support resources can be found at https://www.chef.io/support

Please include as many details about the problem as possible i.e., how to reproduce
the problem (if possible), type of the Operating System and its version, etc.,
and any other relevant details that might help us with troubleshooting.
```

**Comment by jonathanmorley** [comment]
This probably affects more than just omnibus-toolchain

**Comment by wrightp** [comment]
The line was not removed. There is now a condition to check whether an unstable artifact is being queried. I can see by the download URL and metadata that the logic is 2.0.1 is the desired behavior. I will take as to why the `install_file` function switch isn't detecting the dmg extension.

**Comment by wrightp** [comment]
I take that back. I see what like you are referencing. That looks like a a greedy line delete. Fixing now and I'll get a release out today.

**Comment by wrightp** [comment]
fixed in 2.0.2
